### ISSUE TITLE
Pass http auth credentials via click fail

### ISSUE THREAD

**Issue by LuvDeluxe** [cypress-io/cypress#4633]
### Desired behavior:
When a user is on a website with http auth and clicks on a link that is from the same domaine, the credentials are passed if they were specified before.

### Steps to reproduce: (app code and test code)

let searchQueryPagesRedirect = 'cats'
let searchQueryTopicsRedirect = 'dogs'

describe('Opens the new pages redirects, checks if the input values for pages page url redirect correctly', function (){
  it('Open the new pages redirects page', function () {
    let url = localhost
    cy.visit(url, {
      ...conf.HttpAuth(), // This is a custom function for http Auth.
    })
  })

  it('Type a new value in the input box and than update it for new pages redirects', function () {
    cy.get('.page_redirect_input').first().clear().type(searchQueryPagesRedirect)
    cy.get('button[name="action[PageBackNewPageMerge:update]"]').first().click()
  })

  it('Checks if url redirects to the value from the input for new pages redirects', function () {
    cy.get('.page_redirect_container a').first().click() // FAILS because it is not being able to get //httpAhuth In the previous version this was not present.
    cy.url().should('include', searchQueryPagesRedirect)
    cy.get('.xdebug-error').should('not.exist')
  })
})

### Versions
Cypress 3.3.2, MacOS Mojave 10.14.3, Chrome 75.0.3770.100

**Comment by flotwig** [comment]
Hey @LuvDeluxe, a couple of questions:

1. Are you trying to use HTTP Basic Authentication, or another type of auth?
2. Did this work before 3.3.2?
3. What is the URL that you are originally cy.visiting and what is the URL that you end up on that should still have auth?

**Comment by LuvDeluxe** [comment]
@flotwig 

1) I am trying to use HTTP Basic Authentication.
2) Yes, this worked well on 3.3.1 version. I've noticed after the update,  is not working well like before.
3) The original url is: http://back.dev.alive-vitamins.com/back/new-pages-redirects and the destination is: http://dev.alive-vitamins.com/about-us

**Comment by flotwig** [comment]
We released a fix for basic auth in 3.3.2: #4338

Before 3.3.2, we were erroneously sending HTTP Basic Auth credentials to any URL that matched the HTTP origin of the original `cy.visit`ed URL. That's why it worked before - before 3.3.2, sending `auth` to `back.dev.alive-vitamins.com` would make `auth` get sent on all requests to `*.alive-vitamins.com`.

After 3.3.2, we implemented [Protection Spaces according to the RFC](https://tools.ietf.org/html/rfc7235#section-2.2), which state that a protection space is a unique scheme + hostname + port combination. So authenticating to `back.dev.alive-vitamins.com` will only cause authentication to be sent for further requests to `back.dev.alive-vitamins.com`.

This should also match how your browser works, if you want to try visiting those pages within Incognito. I believe Chrome should ask you to authenticate again once you reach `dev.alive-vitamins.com`.
